sw/device/silicon_creator/rom/hooks/README.md

```diff
@@ -0,0 +1,85 @@
+#Silicon Creator ROM Hooks
+
+## Overview
+
+The OpenTitan ROM defines pre-run and post-run hooks for any defined ROM state.
+
+When transitioning to a ROM state, the ROM state machine first executes the pre-run hook.
+
+The state's run callback is called only if the pre-run hook does not return an error.
+Finally, the post-run hook runs upon successful termination of the ROM state's run callback:
+
+```
+│                      │
+│   ┌──────────────┐   │
+│   │Post-run Hook │   │
+│   └───────┬──────┘   │
+└───────────┼──────────┘
+            │
+            │
+┌───────────┼──────────┐
+│ ROM       │          │
+│ State     │          │
+│   ┌───────▼──────┐   │
+│   │ Pre-run Hook │   │
+│   └───────┬──────┘   │
+│           │          │
+│           │          │
+│   ┌───────▽──────┐   │
+│   │ Run Callback │   │
+│   └───────┬──────┘   │
+│           │          │
+│           │          │
+│   ┌───────▽──────┐   │
+│   │Post-run Hook │   │
+│   └───────┬──────┘   │
+└───────────┼──────────┘
+            │
+            │
+┌───────────┼──────────┐
+│   ┌───────▼──────┐   │
+│   │ Pre-run Hook │   │
+│   └──────────────┘   │
+│                      │
+
+```
+
+The transition from one ROM state to the next one is defined by the current ROM state's run callback:
+
+``` c
+rom_error_t rom_state_run_cb(void *arg, rom_state_t *next_state);
+```
+
+By default, the pre-run and post-run hooks do nothing. However, they provide a mechanism for silicon creators to override them with external implementations (e.g. with closed-source, vendor-specific initialization sequences).
+
+Each ROM state pre-run and post-run hooks are defined as weak symbols using a pre-defined naming scheme.
+
+For example the hooks prototypes for a ROM state named `kRomStateInit` would be:
+
+``` c
+OT_WARN_UNUSED_RESULT rom_error_t rom_state_pre_kRomStateInit(void *arg);
+OT_WARN_UNUSED_RESULT rom_error_t rom_state_post_kRomStateInit(void *arg);
```

Comment:
This is different from the example in `hooks/dummy_rom_hooks.c`, which uses a macro to declare the hooks (although the result is the same). Maybe worth clarifying that the macros are an option just to avoid the mismatch?

Reply:
Good catch, thanks. I added some clarification in this file now. Let me know if that looks good to you.